### PR TITLE
feature/us10.2-new-changes-coverage

### PR DESCRIPTION
**Pull request by viktorbll**
US 10.2 Coverage Plugin - Coverage of new methods
US 10.5 Class coverage - Coverage by classes

### CODE REVIEW

File: plugins/drill-coverage-plugin/src/adminPartTest/kotlin/CoverageControllerTest.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    @Test
+    fun `should add class bytes on receiving a CLASS_BYTES message`() {
+        val clazz = Dummy::class.java
+        val bytes = clazz.readBytes()
+        prepareClasses(clazz)
+
+        coverageController.apply {
+            val path = clazz.path
+            assertNotNull(initialClassBytes[path])
+            assertTrue { initialClassBytes[path]!!.contentEquals(bytes) }
```
Comment (IgorKey) [inline plugins/drill-coverage-plugin/src/adminPartTest/kotlin/CoverageControllerTest.kt]: may be swap this 2 args? "!!" is awful :) bytes.contentEquals(initialClassBytes[path])

File: plugins/drill-coverage-plugin/src/agentPartMain/kotlin/CoverageAgentPart.kt [hunk 2]
```diff
@@ -41,17 +42,25 @@ class CoveragePlugin(override val id: String) : InstrumentedPlugin<CoverConfig,
...
-        try {
-            initialClassBytes[className] = initialBytest
-            sendClass(ClassBytes(className, initialBytest.toList()))
-            return instrumenter(className, initialBytest)
-        } catch (ex: Throwable) {
-            ex.printStackTrace()
-            throw ex
-        }
+    override fun instrument(className: String, initialBytes: ByteArray): ByteArray {
+        return if ("_\$\$_" !in className && "CGLIB\$\$" !in className) {
```
Comment (IgorKey) [inline plugins/drill-coverage-plugin/src/agentPartMain/kotlin/CoverageAgentPart.kt]: Why did we skip these classes? There are a lot of ways to do javaProxies. We don't have any chance to predict this. We should extract this to config.

File: plugins/drill-coverage-plugin/src/adminPartTest/kotlin/CoverageControllerTest.kt (hunk 1 above)
Comment (viktorbll) [reply plugins/drill-coverage-plugin/src/adminPartTest/kotlin/CoverageControllerTest.kt]: Agree. The expected value should come first. Fixed it.

File: plugins/drill-coverage-plugin/src/agentPartMain/kotlin/CoverageAgentPart.kt (hunk 2 above)
Comment (viktorbll) [reply plugins/drill-coverage-plugin/src/agentPartMain/kotlin/CoverageAgentPart.kt]: This is a quick solution for the demo. Let's do the config feature in the future.

**Pull request merged**